pages/_app.tsx

```diff
@@ -1,6 +1,55 @@
-import '@/styles/globals.css'
-import type { AppProps } from 'next/app'
+import type { AppProps } from "next/app";
+import { useState, useEffect } from "react";
+import { getUserData } from "./api/SharedApi";
+import Header from "../components/commons/Header";
+import Footer from "../components/commons/Footer";
+import GlobalStyle from "../styles/GlobalStyles";
+import { UserInfo } from "../types/common";
+import useModal from "../hook/useModal";
+import FolderAddInputModal from "../components/commons/modals/FolderAddInputModal";
+import FolderNameChangeModal from "../components/commons/modals/FolderNameInputModal";
+import FolderDeleteModal from "../components/commons/modals/FolderDeleteModal";
+import LinkDeleteModal from "../components/commons/modals/LinkDeleteModal";
+import FolderSharedModal from "../components/commons/modals/FolderSharedModal";
+import LinkAddModal from "../components/commons/modals/LinkAddModal";
 
 export default function App({ Component, pageProps }: AppProps) {
-  return <Component {...pageProps} />
+  const { modal, openModal, closeModal } = useModal();
+  const [user, setUser] = useState<UserInfo>({
+    id: 1,
+    name: "",
+    email: "",
+    profileImageSource: "",
+  });
+
+  const handleEmailLoad = async () => {
+    const data = await getUserData();
+    setUser(data);
+  };
+
+  useEffect(() => {
+    handleEmailLoad();
+  }, []);
```

Comment:
이런 표현도 가능해서 참고로 알려드립니다. 동작이 완전히 같지는 않아요
```suggestion
  useEffect(() => {
    getUserData().then(setUser)
  }, []);
```